zjit/src/hir.rs

```diff
@@ -5297,4 +5421,121 @@ mod opt_tests {
               Return v2
         "#]]);
     }
+
+    #[test]
+    fn test_opt_hash_freeze() {
```

Comment:
A couple of test nits:

a) Can we please do tests that are descriptive of what they're testing? For example, `test_elide_freeze_with_frozen_hash` or something similar
b) Can you please also test that we don't do this optimization if there are args present